docs/src/content/hardhat-runner/docs/guides/typescript.md

```diff
@@ -10,188 +10,131 @@ Hardhat will automatically enable its TypeScript support if your config file end
+:::tab{value=yarn}
 
-export default {
-  solidity: "0.7.3"
-};
+```
+yarn add --dev chai @types/node @types/mocha @types/chai
 ```
 
-You also need to create a [`tsconfig.json`](https://www.typescriptlang.org/docs/handbook/tsconfig-json.html) file. Here's a template you can base yours on:
+:::
 
-```json
-{
-  "compilerOptions": {
-    "target": "es2018",
-    "module": "commonjs",
-    "strict": true,
-    "esModuleInterop": true,
-    "outDir": "dist"
-  },
-  "include": ["./scripts", "./test"],
-  "files": ["./hardhat.config.ts"]
-}
-```
+::::
 
-You can use different settings, but please make sure your Hardhat config file is included. The easiest way of doing this is by keeping its path in the `files` array.
+### TypeScript configuration
 
-And that's really all it takes. Now you can write your config, tests, tasks and scripts in TypeScript.
+You can easily turn a JavaScript Hardhat config file into a TypeScript one. Let's see how this is done starting with a fresh Hardhat project.
 
-## Writing tests and scripts in TypeScript
+Open your terminal, go to an empty folder, run `npx hardhat`, and go through the steps to create a JavaScript project. When you're done your project directory should look something like this:
 
-To write your smart contract tests and scripts you'll most likely need access to an Ethereum library to interact with your smart contracts. This will probably be one of [hardhat-ethers](https://github.com/nomiclabs/hardhat/tree/master/packages/hardhat-ethers) or [hardhat-web3](https://github.com/nomiclabs/hardhat/tree/master/packages/hardhat-web3), all of which inject instances into the [Hardhat Runtime Environment].
+```
+$ ls -l
+total 1200
+drwxr-xr-x    3 pato  wheel      96 Oct 20 12:50 contracts/
+-rw-r--r--    1 pato  wheel     567 Oct 20 12:50 hardhat.config.js
+drwxr-xr-x  434 pato  wheel   13888 Oct 20 12:52 node_modules/
+-rw-r--r--    1 pato  wheel  604835 Oct 20 12:52 package-lock.json
+-rw-r--r--    1 pato  wheel     460 Oct 20 12:52 package.json
+drwxr-xr-x    3 pato  wheel      96 Oct 20 12:50 scripts/
+drwxr-xr-x    3 pato  wheel      96 Oct 20 12:50 test/
+```
 
-When using JavaScript, all the properties in the HRE are injected into the global scope, and are also available by getting the HRE explicitly. When using TypeScript nothing will be available in the global scope and you will need to import everything explicitly.
+Then, you should follow the steps mentioned in the [Installing dependencies](#installing-dependencies) section above.
 
-An example for tests:
+Now, we are going to rename the config file from `hardhat.config.js` to `hardhat.config.ts`, just run:
 
-```typescript
-import { ethers } from "hardhat";
-import { Signer } from "ethers";
+```
+mv hardhat.config.js hardhat.config.ts
+```
 
-describe("Token", function () {
-  let accounts: Signer[];
+We need to a single change to your config for it to work with TypeScript: plugins must be loaded with `import` instead of `require`.
```

Comment:
Is this correct? You also need to change the `module.exports` to `export default`, so it's not a single change.

Reply:
You are right!